datalad/support/archive_utils_patool.py

```diff
@@ -0,0 +1,213 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""patool based implementation for datalad.support.archives utilities"""
+
+import patoolib
+from .external_versions import external_versions
+# There were issues, so let's stay consistently with recent version
+assert(external_versions["patoolib"] >= "1.7")
+
+import os
+from .exceptions import MissingExternalDependency
+from .path import (
+    basename,
+    join as opj,
+    exists,
+)
+
+from datalad.utils import (
```

Comment:
While at it make into a relative import? Note that there are more imports from ..utils below